src/k-provider/ovp/provider.js

```diff
@@ -25,12 +26,25 @@ export default class OVPProvider extends BaseProvider<OVPProviderMediaInfoObject
     OVPConfiguration.set(options.env);
     this._setFilterOptionsConfig(options.filterOptions);
     this._networkRetryConfig = Object.assign(this._networkRetryConfig, options.networkRetryParameters);
+    this._anonymousKs = '';
   }
 
   get env() {
     return OVPConfiguration.get();
   }
 
+  get anonymousKs(): string {
+    return this._anonymousKs;
+  }
+
+  set anonymousKs(value: string): void {
+    this._anonymousKs = value;
+  }
+
+  get userKs(): string {
+    return this.ks || this._anonymousKs;
```

Comment:
this ix very counter intuitive, either we get user KS or we use anonymous KS, so the semantics of user ks is either signed or anonymous is not clear